test/GroupByTest.cpp

```diff
@@ -728,8 +756,219 @@ TEST_F(GroupByOptimizations, hashMapOptimizationGroupedVariable) {
+  input._values.push_back(std::vector{TC(1.0), TC("C")});
+  input._values.push_back(std::vector{TC(3.0), TC("g")});
+  input._values.push_back(std::vector{TC(3.0), TC("h")});
+  input._values.push_back(std::vector{TC(3.0), TC("f")});
+  auto qec = ad_utility::testing::getQec();
+  auto values = ad_utility::makeExecutionTree<Values>(qec, input);
+
+  auto groupConcatExpression1 = makeGroupConcatPimpl(varY);
+  auto aliasGC1 = Alias{groupConcatExpression1, varZ};
+
+  auto varW = Variable{"?w"};
+  auto groupConcatExpression2 = makeGroupConcatPimpl(varY, ",");
+  auto aliasGC2 = Alias{groupConcatExpression2, varW};
+
+  GroupBy groupBy{qec, variablesOnlyX, {aliasGC1, aliasGC2}, std::move(values)};
+  auto result = groupBy.getResult();
+  const auto& table = result->idTable();
+
+  auto getId = makeGetId(qec->getIndex());
+  auto d = DoubleId;
+  auto getLocalVocabId = [&result](const std::string& word) {
+    auto value = result->localVocab().getIndexOrNullopt(word);
+    if (value.has_value())
+      return ValueId::makeFromLocalVocabIndex(value.value());
+    else
+      AD_THROW("");
+  };
+
+  auto expected = makeIdTableFromVector(
+      {{d(1), getLocalVocabId("B A C"), getLocalVocabId("B,A,C")},
+       {d(3), getLocalVocabId("g h f"), getLocalVocabId("g,h,f")}});
+  EXPECT_EQ(table, expected);
+
+  RuntimeParameters().set<"use-group-by-hash-map-optimization">(false);
+}
+
+// _____________________________________________________________________________
+TEST_F(GroupByOptimizations, hashMapOptimizationMinMaxIndex) {
+  RuntimeParameters().set<"use-group-by-hash-map-optimization">(true);
+
+  std::string turtleInput =
+      "<x> <label> \"C\" . <x> <label> \"B\" . <x> <label> \"A\" . "
+      "<y> <label> \"g\" . <y> <label> \"f\" . <y> <label> \"h\"";
+
+  QueryExecutionContext* qec = getQec(turtleInput);
+
+  Tree xyScan = makeExecutionTree<IndexScan>(
+      qec, Permutation::Enum::PSO, SparqlTriple{varX, {"<label>"}, varY});
+
+  // Optimization will not be used if subtree is not sort
+  std::vector<ColumnIndex> sortedColumns = {0};
+  Tree subtreeWithSort = makeExecutionTree<Sort>(qec, xyScan, sortedColumns);
+
+  auto minExpression = makeMinPimpl(varY);
+  auto aliasMin = Alias{minExpression, varZ};
+
+  auto varW = Variable{"?w"};
+  auto maxExpression = makeMaxPimpl(varY);
+  auto aliasMax = Alias{maxExpression, varW};
+
+  // SELECT (MIN(?y) as ?z) (MAX(?y) as ?w) WHERE {...} GROUP BY ?x
+  GroupBy groupBy{qec, variablesOnlyX, {aliasMin, aliasMax}, subtreeWithSort};
+  auto result = groupBy.getResult();
+  const auto& table = result->idTable();
+
+  auto getId = makeGetId(qec->getIndex());
+
+  auto expected =
+      makeIdTableFromVector({{getId("<x>"), getId("\"A\""), getId("\"C\"")},
+                             {getId("<y>"), getId("\"f\""), getId("\"h\"")}});
+  EXPECT_EQ(table, expected);
+
+  RuntimeParameters().set<"use-group-by-hash-map-optimization">(false);
+}
+
```

Comment:
One idea I have for a separate PR:

We have (I hope at least) tests for the `aggregate expressions` 
that use a generic framework.
You can extend this, s.t. it also automagically tests the associate AggregationData from this file.
So for example
Evaluating a `SumExpression` or adding all theinputs to a `SumAggregationData` via `addValue` and then finally calling `calculateResult` should always yield the same value.

Then your code immediately also benefits from additionally added unit tests.